server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java

```diff
@@ -412,22 +412,25 @@ protected void populateMail(Message message, MailImpl mail) throws JMSException
         splitter.split(attributeNames)
                 .forEach(name -> setMailAttribute(message, mail, name));
 
-        String sender = message.getStringProperty(JAMES_MAIL_SENDER);
-        if (sender == null || sender.trim().length() <= 0) {
-            mail.setSender(null);
-        } else {
-            try {
-                mail.setSender(new MailAddress(sender));
-            } catch (AddressException e) {
-                // Should never happen as long as the user does not modify the
-                // the header by himself
-                LOGGER.error("Unable to parse the sender address {} for mail {}, so we fallback to a null sender", sender, mail.getName(), e);
-                mail.setSender(null);
-            }
-        }
-
+        mail.setSender(getMailSender(message.getStringProperty(JAMES_MAIL_SENDER), mail));
         mail.setState(message.getStringProperty(JAMES_MAIL_STATE));
+    }
 
+    private MailAddress getMailSender(String sender, Mail mail) {
+        if (sender == null || sender.trim().length() <= 0) {
+            return null;
+        }
+        if (sender.equals(MailAddress.NULL_SENDER_AS_STRING)) {
+            return MailAddress.nullSender();
+        }
+        try {
+            return new MailAddress(sender);
+        } catch (AddressException e) {
+            // Should never happen as long as the user does not modify the
+            // the header by himself
+            LOGGER.error("Unable to parse the sender address {} for mail {}, so we fallback to a null sender", sender, mail.getName(), e);
+            return null;
```

Comment:
can you explain me the difference between a null sender and a NULL_SENDER? (I mean functionally, I understant the null sender is null and NULL_SENDER is <>)

Reply:
I got no idea, I must admit...

SMTP layer does not enqueue `null` but I imagine we could force 'null sender' usage here also...